public/app/core/components/NavBar/NavBar.tsx

```diff
@@ -27,36 +47,141 @@ export const NavBar: FC = React.memo(() => {
+            <Branding.MenuLogo />
+          </NavBarItem>
+          <NavBarItem
+            className={styles.search}
+            isActive={activeItemId === 'search'}
+            label="Search dashboards"
+            onClick={onOpenSearch}
+          >
+            <Icon name="search" size="xl" />
+          </NavBarItem>
+        </NavBarSection>
+      )}
+      {newNavigationEnabled && (
+        <NavBarSection>
+          <NavBarItem url={homeUrl} label="Home" className={styles.grafanaLogo} showMenu={false}>
+            <Branding.MenuLogo />
+          </NavBarItem>
+          <NavBarItem
+            className={styles.search}
+            isActive={activeItemId === 'search'}
+            label="Search dashboards"
+            onClick={onOpenSearch}
+          >
+            <Icon name="search" size="xl" />
+          </NavBarItem>
+        </NavBarSection>
+      )}
+
+      <NavBarSection>
+        {coreItems.map((link, index) => (
+          <NavBarItem
+            key={`${link.id}-${index}`}
+            isActive={activeItemId === link.id}
+            label={link.text}
+            menuItems={link.children}
+            target={link.target}
+            url={link.url}
+          >
+            {link.icon && <Icon name={link.icon as IconName} size="xl" />}
+            {link.img && <img src={link.img} alt={`${link.text} logo`} />}
+          </NavBarItem>
+        ))}
+      </NavBarSection>
+
+      {pluginItems.length > 0 && (
+        <NavBarSection>
+          {pluginItems.map((link, index) => (
+            <NavBarItem
+              key={`${link.id}-${index}`}
+              isActive={activeItemId === link.id}
+              label={link.text}
+              menuItems={link.children}
+              menuSubTitle={newNavigationEnabled ? link.subTitle : undefined}
+              onClick={link.onClick}
+              target={link.target}
+              url={link.url}
+            >
+              {link.icon && <Icon name={link.icon as IconName} size="xl" />}
+              {link.img && <img src={link.img} alt={`${link.text} logo`} />}
+            </NavBarItem>
+          ))}
+        </NavBarSection>
+      )}
+
+      <div className={styles.spacer} />
+
+      <NavBarSection>
+        {configItems.map((link, index) => (
+          <NavBarItem
+            key={`${link.id}-${index}`}
+            isActive={activeItemId === link.id}
+            label={link.text}
+            menuItems={link.children}
+            menuSubTitle={link.subTitle}
+            onClick={link.onClick}
+            reverseMenuDirection
+            target={link.target}
+            url={link.url}
+          >
+            {link.icon && <Icon name={link.icon as IconName} size="xl" />}
+            {link.img && <img src={link.img} alt={`${link.text} logo`} />}
+          </NavBarItem>
+        ))}
+      </NavBarSection>
+
+      {showSwitcherModal && <OrgSwitcher onDismiss={toggleSwitcherModal} />}
```

Comment:
nit: I think I would create 2 separate NavBar components one for the current and one for the feature toggle. I think that makes it a lot cleaner.

Reply:
Agree with this one.